src/game/TileEngine/RenderWorld.cc

```diff
@@ -2478,10 +2478,10 @@ static void Blt8BPPDataTo16BPPBufferTransZIncClip(UINT16* pBuffer, UINT32 uiDest
 	}
 
 	// Calculate rows hanging off each side of the screen
-	const INT32 LeftSkip   = __min(ClipX1 -   MIN(ClipX1, iTempX), usWidth);
-	INT32       TopSkip    = __min(ClipY1 - __min(ClipY1, iTempY), usHeight);
-	const INT32 RightSkip  = __min(  MAX(ClipX2, iTempX + usWidth)  - ClipX2, usWidth);
-	const INT32 BottomSkip = __min(__max(ClipY2, iTempY + usHeight) - ClipY2, usHeight);
+	const INT32 LeftSkip   = std::min(ClipX1 - std::min(ClipX1, iTempX), usWidth);
+	INT32       TopSkip    = std::min(ClipY1 - std::min(ClipY1, iTempY), usHeight);
+	const INT32 RightSkip  = std::clamp(iTempX + usWidth - ClipX2, 0, usWidth);
```

Comment:
I just did some algebra:
```c
__min( MAX(ClipX2, iTempX + usWidth) - ClipX2, usWidth);
__min( MAX(ClipX2 - ClipX2, iTempX + usWidth - ClipX2), usWidth);
__min( MAX(0, iTempX + usWidth  - ClipX2), usWidth);
```

which is then either
```c
__min( iTempX + usWidth  - ClipX2, usWidth); // iff iTempX + usWidth  - ClipX2 > 0
__min( 0, usWidth); // 0
```
so it can't be smaller than 0. And if we look closer at the first case
```c
__min( iTempX + usWidth - ClipX2, usWidth);
__min( iTempX - ClipX2, 0) + usWidth;
```
so it can be at most usWidth. And then the proposed change seems proven to me:

```c
std::clamp(iTempX + usWidth - ClipX2, 0, usWidth);
```

we also know the final values are smaller than usWidth (see a few lines lower) and probably non-negative, but that's just speculation compared to the above.

Reply:
Thanks for explaining.